doc/ref/configuration/master.rst

```diff
@@ -4500,6 +4500,11 @@ strategy between different sources. It accepts 5 values:
 
   Guesses the best strategy based on the "renderer" setting.
 
+.. note::
+    In order for yamlex based features such as ``!aggregate`` to work as expected
+    across documents using the default ``smart`` merge strategy, the :conf_master:`renderer`
+    config option must be set to ``yamlex_jinja`` or similar.
```

Comment:
While this is technically correct, this is a long-outdated method of specifying multiple renderers. We've used a pipe syntax for a while now (the old syntax is translated in `salt/template.py`), so our documentation should suggest using a renderer of `jinja|yamlex`.

For that matter, we should probably change the default renderer in `salt/config/__init__.py` from `yaml_jinja` to `jinja|yaml`, and make the same updates both `doc/ref/configuration/master.rst` and in in the config stubs (`conf/minion`, `conf/master`, et al).

Having said all of that, I'm fine with merging this as-is and making all of the rest of the changes in a future PR.

Reply:
I agree!

Reply:
@mattp- if you would like to change the example to `jinja|yamlex` then I can make all of the other changes I noted above.

Reply:
Otherwise we can just merge this and I can update this reference myself.

Reply:
@terminalmage was going to say I got incepted by the dictupdate code, but I just checked again:
```
105     ¦   if renderer == 'yamlex' or renderer.startswith('yamlex_'):
```
so it seems you actually do need the legacy yamlex_jinja as it stands :)
I'm not actually sure it even makes sense to be relying on the default opts['renderer'] attribute here, but trying to bubble up the pillar/state shebang from the template for each sls seemed ... not easy, as an alternative. any thoughts on how to proceed? guessing renderer.contains('yamlex').

Reply:
also, happy to update the other doc references once decision on ^ is confirmed

Reply:
@mattp- I took care of the other doc references and also rewrote a lot of the renderer docs in #47111. As for the dictupdate code, the proper way to support templated yamlex would be to split on a pipe and confirm that the last element is `yamlex`, which I've also done in #47111. We would still need to keep the second half of the if clause as-is since we need to support the legacy syntax.

If you update line 4506 here to use `jinja|yamlex` instead of `yamlex_jinja`, then that should be all that is needed.